File: src/features/typing/data/java/two_sum.java
import java.util.HashMap;
import java.util.Map;

class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> seen = new HashMap<>();
        for (int i = 0; i < nums.length; i++) {
            if (seen.containsKey(target - nums[i])) {
                return new int[] { seen.get(target - nums[i]), i };
            }
            seen.put(nums[i], i);
        }
        return new int[] {};
    }
}

File: src/features/typing/data/dart/two_sum.dart
class Solution {
  List<int> twoSum(List<int> nums, int target) {
    final seen = <int, int>{};
    for (int i = 0; i < nums.length; i++) {
      final complement = target - nums[i];
      if (seen.containsKey(complement)) {
        return [seen[complement]!, i];
      }
      seen[nums[i]] = i;
    }
    return [];
  }
}
